presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java

```diff
@@ -88,6 +88,9 @@ public CassandraPartitionResult getPartitions(ConnectorTableHandle tableHandle,
                 List<ColumnHandle> partitionColumns = (List) partitionKeys;
                 remainingTupleDomain = TupleDomain.withColumnDomains(Maps.filterKeys(tupleDomain.getDomains().get(), not(in(partitionColumns))));
                 List<ColumnHandle> clusteringColumns = (List) table.getClusteringKeyColumns();
+                if (!getPartitionKeysSet(table, tupleDomain).isEmpty()) { // Prune clustering keys predicates only if there is a predicate on partition keys too
```

Comment:
So you should only add the predicate in the cassandra query if there's a predicate on partition keys too